packages/fuels/tests/harness.rs

```diff
@@ -3596,3 +3602,62 @@ async fn testnet_hello_world() -> Result<(), Error> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_input_message() -> Result<(), Error> {
+    abigen!(
+        MyContract,
+        "packages/fuels/tests/test_projects/contract_test/out/debug/contract_test-abi.json"
+    );
+
+    let compare_messages =
+        |messages_from_provider: Vec<OtherMessage>, used_messages: Vec<Message>| -> bool {
+            zip(&used_messages, &messages_from_provider).all(|(a, b)| {
+                a.sender == b.sender.0 .0
+                    && a.recipient == b.recipient.0 .0
+                    && a.owner == b.owner.0 .0
+                    && a.nonce == b.nonce.0
+                    && a.amount == b.amount.0
```

Comment:
Maybe we could extract this comparison logic somewhere outside of a single test? I don't see it being used elsewhere, but it's still not very intuitive (for me).

Reply:
It was outside, but since we don't use it anywhere else, I put it inside.